services/products-service/lambdas/catalogBatchProcess.ts

```diff
@@ -0,0 +1,58 @@
+import { SQSEvent } from 'aws-lambda';
+import { SNSClient, PublishCommand } from '@aws-sdk/client-sns'; // ES Modules import
+
+import * as AWS from '@aws-sdk/client-dynamodb';
+import {
+  DynamoDBDocumentClient,
+  TransactWriteCommand,
+} from '@aws-sdk/lib-dynamodb';
+
+const snsClient = new SNSClient();
+const dbClient = new AWS.DynamoDB();
+const ddbDocClient = DynamoDBDocumentClient.from(dbClient);
+
+const handler = async (event: SQSEvent) => {
+  try {
+    console.log('catalogBatchProcess event', JSON.stringify(event, null, 2));
+
+    await Promise.all(
+      event.Records.map(async (record) => {
+        console.log('SQS record', JSON.stringify(record, null, 2));
+
+        const body = JSON.parse(record.body);
+        const { product_id, title, description, count, price } = body;
```

Comment:
It would be great to have body validation here before storing an item in DynamoDB.